gapic/src/main/com/google/gapid/perfetto/views/State.java

```diff
@@ -55,8 +55,10 @@
   private double maxScrollOffset = 0;
   private double nanosPerPx;
   private long resolution;
-  private Selection selection;
+  private Selection.MultiSelection selection;
   private final AtomicInteger lastSelectionUpdateId = new AtomicInteger(0);
+  private long selectedUpid = -1;   // For a selected CPU slice.
+  private long selectedUtid = -1;   // For a selected CPU slice.
```

Comment:
Just `private ThreadInfo selectedThread`. The `ThreadInfo` keeps track of the `utid` and `upid` for you.